src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -125,6 +125,14 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
             //     V v = (V)(T)e.Current;
             //     /* node.Body */
             // }
+
+            // The scope of the iteration variable is the foreach statement syntax.
```

Comment:
Also, not that a BoundBlock for foreach statement iterating over IEnumerable already is associated with the foreach statement (the rewriter takes a different code path). So if we decide to go for 2) we will also need to make a change in the other code path.